dot/state/storage.go

```diff
@@ -107,6 +119,26 @@ func (s *StorageState) StoreTrie(ts *rtstorage.TrieState) error {
 	s.tries[root] = ts.Trie()
 	s.lock.Unlock()
 
+	if _, ok := s.pruner.(*fullNodePruner); header == nil && ok {
+		return fmt.Errorf("block cannot be empty for Full node pruner")
+	}
+
+	if header != nil {
```

Comment:
shouldn't this case be if the pruner is a `*fullNodePruner`? I think the code should never pass a nil header

Reply:
In testing, we are passing nil blocks for *archivalNodePruner.